src/notification.c

```diff
@@ -253,7 +253,15 @@ markup_parser_text (GMarkupParseContext  *context,
 {
   GString *composed = user_data;
 
-  g_string_append_len (composed, text, text_len);
+  while (text_len > 0)
+    {
+      gsize len = MIN (text_len, G_MAXSSIZE);
+      g_autofree char *escaped = g_markup_escape_text (text, len);
+
+      g_string_append (composed, escaped);
+      text_len -= len;
+      text += len;
+    }
```

Comment:
This loop is because strictly speaking there is no API guarantee that `text_len` won't exceed `G_MAXSSIZE`, but `g_markup_escape_text()` can "only" escape text in chunks of length no more than `G_MAXSSIZE` bytes.

(On 32-bit platforms, if you have somehow managed to send a notification with more than 2 GiB of text... well done, I suppose? And on 64-bit platforms this clearly can't actually happen, but writing the code in a way that defends against it seemed easier than justifying why it can't.)